src/az_ulib_ipc/az_ulib_ipc.c

```diff
@@ -686,6 +710,133 @@ AZ_NODISCARD az_result az_ulib_ipc_call_with_str(
   return result;
 }
 
+AZ_NODISCARD az_result az_ulib_ipc_split_method_name(
+    az_span full_name,
+    az_span* device_name,
+    az_span* package_name,
+    uint32_t* package_version,
+    az_span* interface_name,
+    uint32_t* interface_version,
+    az_span* capability_name)
+{
+  _az_PRECONDITION_NOT_NULL(_az_ipc_cb);
+  _az_PRECONDITION_VALID_SPAN(full_name, 1, false);
+  _az_PRECONDITION_NOT_NULL(device_name);
+  _az_PRECONDITION_NOT_NULL(package_name);
+  _az_PRECONDITION_NOT_NULL(package_version);
+  _az_PRECONDITION_NOT_NULL(interface_name);
+  _az_PRECONDITION_NOT_NULL(interface_version);
+  _az_PRECONDITION_NOT_NULL(capability_name);
+
+  AZ_ULIB_TRY
+  {
+    uint32_t package_version_temp;
+    int32_t interface_name_session;
+    uint32_t interface_version_temp;
+    bool has_capability_name;
+    int32_t count_session = 0;
+    az_span sessions[MAX_SESSIONS_IN_METHOD_FULL_NAME];
```

Comment:
The name "sessions" makes me think of network connections or maybe references to an object. Can you rename to something like "names", "fields", or "descriptors"?

Reply:
I like fields.